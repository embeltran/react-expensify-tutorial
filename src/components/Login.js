import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const Login = ({ startLogin }) => (
  <div className="box-layout"> 
    <div>
      <h2>expensify app</h2>
      <p>tag line</p>      
      <button onClick={startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);