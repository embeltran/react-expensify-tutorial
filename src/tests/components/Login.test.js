import React from 'react';
import { shallow } from 'enzyme';
import Login from '../../components/Login';

test('should render Login correctly', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
});
