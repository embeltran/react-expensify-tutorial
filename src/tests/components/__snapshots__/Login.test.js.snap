// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login correctly 1`] = `
<div
  className="box-layout"
>
  <div>
    <h2>
      expensify
    </h2>
    <p>
      tag line
    </p>
    <button>
      Login with Google
    </button>
  </div>
</div>
`;
